binary_search_tree/binary_search_tree.py

```diff
@@ -3,20 +3,92 @@
 from dll_queue import Queue
 from dll_stack import Stack
 
+# Questions:
+# Only ints? 
+# Negative numbers?
+
+# Observations
+# >= goes right
+# Need to traverse to delete
+# When deleting, the smallest child becomes parent
+
+
 class BinarySearchTree:
-  def __init__(self, value):
+  def __init__(self, value): # We're just using value, so key is value
     self.value = value
     self.left = None
     self.right = None
+    self.size = 0
 
+  # * `insert` adds the input value to the binary search tree, adhering to the
+  # rules of the ordering of elements in a binary search tree.
+  # Need to traverse to find spot to insert
   def insert(self, value):
-    pass
+    # if value exists, return
+    if value == self.value:
+      return
+    # if value is less than self.value...
+    elif value < self.value:
+      # if it's self.left...
+      if self.left:
+        # insert the value to the left
+        self.left.insert(value)
+      else:
+        # else, recurse
+        self.left = BinarySearchTree(value)
+    else:
+      # if it's self.right...
+      if self.right:
+        # insert the value to the right
+        self.right.insert(value)
+      else:
+        # else, recurse
+        self.right = BinarySearchTree(value)
 
+  # * `contains` searches the binary search tree for the input value, 
+  # returning a boolean indicating whether the value exists in the tree or not.
+  # Start from root and traverse the tree
+  # We can stop at the first instance of a value
+  # We know it's not found if we get to a node that doesn't have children
   def contains(self, target):
-    pass
+    # if self.value = target, return True
+    if self.value == target:
+      return True
+    # if target is less than self.value...
+    elif target < self.value:
+      #if it is self.left...
+      if self.left:
+        # recurse
+        return self.left.contains(target)
```

Comment:
I don't think you need to return this (here or below) but you do need to invoke `self.left.contains(target)` so maybe just a different way to do it :)